DatadogCore/Sources/Core/Storage/Reading/Reader.swift

```diff
@@ -24,6 +24,6 @@ extension Batch {
 
 /// A type, reading batched data.
 internal protocol Reader {
-    func readNextBatch() -> Batch?
+    func readNextBatches(_ count: Int?) -> [Batch]
```

Comment:
If `count` is optional, what does `count: nil` mean in the context of retrieving next batches? It deserves an API comment with explanation.